daemon/daemon.go

```diff
@@ -744,6 +745,7 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 }
 
 func (daemon *Daemon) Shutdown() error {
+	daemon.shutdown = true
```

Comment:
Do you mean wrong behavior when running stop after shutdown is in progress or just protecting with atomic.